contracts/Puzzle.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity >=0.4.22; 
+
+contract GameContract {
+    string internal constant INSUFFICIENT_FUND_MESSAGE = "Insufficient Fund";
+    string internal constant RESTRICTED_MESSAGE = "Unauthorized Access";
+    string internal constant INCORRECT_SESSION = "Player requesting payout for unknown session";
+    string internal constant INCORRECT_LEVEL = "Player requesting payout for earlier level";
+    uint amount;
+    uint sessionID;
+    uint progress;
+    string seq;
+    uint constant thresholdLevel = 4;
+    mapping(address => uint) playerLevel;
+    mapping(address => uint) playerSession;
+    mapping(address => uint) playerStake;
+    address public manager;  // The adress of the owner of this contract
+    address payable[] public players;   // The players of current session
+
+    constructor() public {
+        manager = msg.sender;
+    }
+    
+    /**
+     * @dev random number generator
+     *     
+     */
+    function random() private returns (uint) {
+        sessionID = uint(keccak256(abi.encodePacked(now, msg.sender, this)));
+        return sessionID;   
+    }
+    
+    /**
+     * @dev The player enters into the current game session by
+     *      paying at least 2 token.
+     */
+    function play() public payable returns (uint) {
+        require(msg.value >= 2 ether, INSUFFICIENT_FUND_MESSAGE);
+        sessionID = random();
+        playerLevel[msg.sender] = 0;
+        playerSession[msg.sender] = sessionID;
+        playerStake[msg.sender] = msg.value;
+        players.push(msg.sender);
+        return sessionID;
```

Comment:
is sessionID the txid?

Reply:
No, its just a randomness we generate to maintain a session.